core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -4176,7 +4176,7 @@ llvm_jit_invoke_native(WASMExecEnv *exec_env, uint32 func_idx, uint32 argc,
 #if WASM_ENABLE_BULK_MEMORY != 0
 bool
 llvm_jit_memory_init(WASMModuleInstance *module_inst, uint32 seg_index,
-                     uint32 offset, uint32 len, uint32 dst)
+                     uint32 offset, uint32 len, size_t dst)
```

Comment:
Should also call `clamp_u64_to_u32(memory_inst->memory_data_size - dst)` in below `bh_memcpy_s`

Reply:
I haven't looked into JIT and there's no way to test memory64 in JIT right now as memory64 is not supported in this mode, but that's a reasonable change, updated.